### PR TITLE
Remove Python 3.6

### PR DESCRIPTION
**Pull request by Kludex**
Python 3.6 EOL is 23 Dec 2021, and this PR removes it. 

References: 
- https://endoflife.date/python
- https://pypistats.org/packages/uvicorn

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 **Documentation**: [https://www.uvicorn.org](https://www.uvicorn.org)
 
-**Requirements**: Python 3.6+ (For Python 3.5 support, install version 0.8.6.)
+**Requirements**: Python 3.7+ (For Python 3.6 support, install version 0.15.0.)
```
Comment (Kludex) [inline README.md:16]: I've removed the 3.5 info. ~15 daily downloads for Python 3.5 and this information is only relevant if you want to start using uvicorn on a Python 3.5 project.

Reference: https://pypistats.org/packages/uvicorn

**Comment by euri10** [comment]
lgtm, would be cool if @graingert can take a quick look since most deletions are from stuff he wrote iirc !

**Review by lovelydinosaur — APPROVED** [review]
Nice. 👍

**Comment by Kludex** [comment]
Thanks!

**Pull request merged**